File: buzzvil-sdk-ios/buzzvil-sdk-ios-objc/Feed/Custom/CustomFeedHeaderView.m
#import "CustomFeedHeaderView.h"

@interface CustomFeedHeaderView ()

@property (nonatomic, strong, readonly) UILabel *headerLabel;

@end

@implementation CustomFeedHeaderView

+ (CGFloat)desiredHeight {
  return 60;
}

- (instancetype)initWithCoder:(NSCoder *)coder {
  if (self = [super initWithCoder:coder]) {
    [self setupView];
  }
  return self;
}

- (instancetype)initWithFrame:(CGRect)frame {
  if (self = [super initWithFrame:frame]) {
    [self setupView];
  }
  return self;
}

- (void)setupView {
  _headerLabel = [[UILabel alloc] initWithFrame:CGRectZero];
  [self addSubview:_headerLabel];

  // LayoutConstraint 설정
  // ...
  _headerLabel.translatesAutoresizingMaskIntoConstraints = NO;
  [NSLayoutConstraint activateConstraints:@[
    [_headerLabel.trailingAnchor constraintEqualToAnchor:self.trailingAnchor],
    [_headerLabel.centerYAnchor constraintEqualToAnchor:self.centerYAnchor],
  ]];
}

- (void)availableRewardDidUpdate:(NSInteger)reward {
  _headerLabel.text = [NSString stringWithFormat:@"리워드 %ld원", reward];
}

@end

File: buzzvil-sdk-ios/buzzvil-sdk-ios-swift/Feed/Custom/CustomFeedHeaderView.swift
import UIKit
import BuzzvilSDK

final class CustomFeedHeaderView: BZVFeedHeaderView {
  let headerLabel = UILabel(frame: .zero)
  
  override class func desiredHeight(_ width: CGFloat) -> CGFloat {
    return 100.0
  }

  required init?(coder: NSCoder) {
    super.init(coder: coder)
    setUpView()
  }

  override init(frame: CGRect) {
    super.init(frame: frame)
    setUpView()
  }

  func setUpView() {
    addSubview(headerLabel)
    headerLabel.textColor = .white
    headerLabel.text = "HEADER CUSTOM"

    // LayoutConstraint 설정
    // ...
    headerLabel.translatesAutoresizingMaskIntoConstraints = false
    NSLayoutConstraint.activate([
      headerLabel.trailingAnchor.constraint(equalTo: trailingAnchor),
      headerLabel.centerYAnchor.constraint(equalTo: centerYAnchor),
    ])
  }
  
  override func availableRewardDidUpdate(_ reward: Int) {
    headerLabel.text = "리워드 \(reward)원"
  }
}
